### PR TITLE
Allows prison jumpsuits to be reskinned, complimenting alternative titles for prisoners

### PR DESCRIPTION
**Pull request by NopemanMcHalt**
# About The Pull Request
Adds color-coded prison jumpsuit as reskins. Idea is borrowed from Prison Architect (at least, before Paradox's acquirement).

## Why It's Good For The Game
So officers can enforce and change jumpsuit depending on prisoner behavior or their charges.

## A Port?
Ported from ye olde [Skyrat13](https://github.com/Skyrat-SS13/Skyrat13).
Black jumpsuit/"Vampire" variant is done by Tony Meowprano#2038 (i forgor:skull: to note it in commit because of being a hurry-ass).

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl: Anonymous
add: Prison jumpsuits can be re-skinned into other color-coded security level, including protective custody and "vampire" variant.
/:cl:

### CODE REVIEW

File: modular_splurt/code/modules/clothing/under/miscellaneous.dm [hunk 1]
```diff
@@ -341,3 +341,34 @@
...
+	. = ..()
+	name = "prison [current_skin] jumpsuit"
+
+/obj/item/clothing/under/rank/prisoner/skirt
+	unique_reskin = list(
+		"max-sec"			= list("icon_state" = "maxprisoner_skirt"),
+		"high-sec"			= list("icon_state" = "highprisoner_skirt"),
+		"med-sec"			= list("icon_state" = "prisoner_skirt"),
+		"low-sec"			= list("icon_state" = "lowprisoner_skirt"),
+		"prot-sec"			= list("icon_state" = "protprisoner_skirt")
```
[the remark this answers is not in the document]
Comment (NopemanMcHalt) [reply modular_splurt/code/modules/clothing/under/miscellaneous.dm:369]: That's sadism but I'll try it.

[the remark this answers is not in the document]
Comment (NopemanMcHalt) [reply modular_splurt/code/modules/clothing/under/miscellaneous.dm]: Nevermind it worked. Reskin button is kept to be CTRL.

File: modular_splurt/code/modules/clothing/under/miscellaneous.dm [hunk 2]
```diff
@@ -341,3 +341,31 @@
...
 	mob_overlay_icon = 'modular_splurt/icons/mob/clothing/uniform.dmi'
 	mutantrace_variation = STYLE_DIGITIGRADE|STYLE_NO_ANTHRO_ICON
+
+// Prison Jumpsuit/Jumpskirt override to allow reskins
+/obj/item/clothing/under/rank/prisoner
+	reskin_binding = COMSIG_CLICK_CTRL
+	unique_reskin = list(
+		"max-sec"			= list("icon_state" = "maxprisoner", "icon" = 'modular_splurt/icons/obj/clothing/uniforms.dmi', "mob_overlay_icon" = 'modular_splurt/icons/mobs/clothing/uniform.dmi', "anthro_mob_worn_overlay" = 'modular_splurt/icons/mobs/clothing/uniform_digi.dmi'),
+		"high-sec"			= list("icon_state" = "highprisoner", "icon" = 'modular_splurt/icons/obj/clothing/uniforms.dmi', "mob_overlay_icon" = 'modular_splurt/icons/mobs/clothing/uniform.dmi', "anthro_mob_worn_overlay" = 'modular_splurt/icons/mobs/clothing/uniform_digi.dmi'),
+		"med-sec"			= list("icon_state" = "prisoner"),
```
Comment (BongaTheProto) [inline modular_splurt/code/modules/clothing/under/miscellaneous.dm:351]:
```suggestion
		"med-sec"			= list("icon_state" = "prisoner", "icon" = 'icons/obj/clothing/uniforms.dmi', "mob_overlay_icon" = null, "anthro_mob_worn_overlay" = null),
```

File: modular_splurt/code/modules/clothing/under/miscellaneous.dm [hunk 3]
```diff
@@ -341,3 +341,31 @@
...
+
+/obj/item/clothing/under/rank/prisoner/reskin_obj(mob/M)
+	. = ..()
+	name = "prison [current_skin] jumpsuit"
+
+/obj/item/clothing/under/rank/prisoner/skirt
+	unique_reskin = list(
+		"max-sec"			= list("icon_state" = "maxprisoner_skirt", "icon" = 'modular_splurt/icons/obj/clothing/uniforms.dmi', "mob_overlay_icon" = 'modular_splurt/icons/mobs/clothing/uniform.dmi'),
+		"high-sec"			= list("icon_state" = "highprisoner_skirt", "icon" = 'modular_splurt/icons/obj/clothing/uniforms.dmi', "mob_overlay_icon" = 'modular_splurt/icons/mobs/clothing/uniform.dmi'),
+		"med-sec"			= list("icon_state" = "prisoner_skirt"),
```
Comment (BongaTheProto) [inline modular_splurt/code/modules/clothing/under/miscellaneous.dm:364]:
```suggestion
		"med-sec"			= list("icon_state" = "prisoner_skirt",  "icon" = 'icons/obj/clothing/uniforms.dmi', "mob_overlay_icon" = null, "anthro_mob_worn_overlay" = null),
```

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

**Pull request merged**